addons/project/controllers/portal.py

```diff
@@ -35,7 +35,7 @@ def _project_get_page_view_values(self, project, access_token, page=1, date_begi
         domain = [('project_id', '=', project.id)]
         # pager
         url = "/my/projects/%s" % project.id
-        values = self._prepare_tasks_values(page, date_begin, date_end, sortby, search, search_in, groupby, url, domain, su=bool(access_token), project=project)
+        values = self._prepare_tasks_values(page, date_begin, date_end, sortby, search, search_in, groupby, url, domain, su=request.env.su, project=project)
```

Comment:
```suggestion
        values = self._prepare_tasks_values(page, date_begin, date_end, sortby, search, search_in, groupby, url, domain, su=bool(access_token) and request.env.user.has_group('base.group_public'), project=project)
```

By doing that, we will only do in sudo if the public user open the url with an access token. If the public user has not the url with the access token then it will have an access error since he has no access to project model.